test/utils.test.ts

```diff
@@ -480,14 +480,89 @@ describe('getExpectedExecutionTimestamp', () => {
 
       expect(expectedExeuction).to.be.equal(nextUpdate + updateInterval);
     });
+    it('oldCommits', () => {
+      //                          | now
+      //                          | lastUpdate
+      //                     | updateBeforeThat
+      //               | commitCreated
+      let lastUpdate = now; // pool was just updated
+      let oldCommit = commitCreated - (2 * updateInterval)
+      let nextUpdate = lastUpdate + updateInterval;
+
+      let expectedExeuction = getExpectedExecutionTimestamp(
+        frontRunningInterval,
+        updateInterval,
+        lastUpdate,
+        oldCommit 
+      )
+      expect(expectedExeuction).to.be.equal(lastUpdate - updateInterval);
+
+      //                          | now
+      //                           | lastUpdate
+      //                      | updateBeforeThat
+      //                | commitCreated
+      lastUpdate = now + 1;
+      nextUpdate = lastUpdate + updateInterval;
+
+      expectedExeuction = getExpectedExecutionTimestamp(
+        frontRunningInterval,
+        updateInterval,
+        lastUpdate,
+        oldCommit 
+      )
+      expect(expectedExeuction).to.be.equal(lastUpdate - updateInterval);
+
+      oldCommit = commitCreated - updateInterval;
+      expectedExeuction = getExpectedExecutionTimestamp(
+        frontRunningInterval,
+        updateInterval,
+        lastUpdate,
+        oldCommit 
+      )
+      expect(expectedExeuction).to.be.equal(nextUpdate - updateInterval); // should have already been executed
```

Comment:
I would have thought this should be `lastUpdate - (2 * updateInterval) + 1` since its getting shifted back in time by `updateInterval`